### PR TITLE
Wheel buttons dev

### PR DESCRIPTION
**Pull request by bracyw**
## Changes

_Explanation of changes goes here_

## Notes

_Any other notes go here_

## Test Cases

- Case A
- Edge case
- ...

## To Do

_Any remaining things that need to get done_

- [ ] item 1
- [ ] ...

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please reach out to your Project Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [ ] All commits are tagged with the ticket number
- [ ] No merge conflicts
- [ ] All checks passing
- [ ] Remove any non-applicable sections of this template
- [ ] Assign the PR to yourself
- [ ] Request reviewers & ping on Slack
- [ ] PR is linked to the ticket (fill in the closes line below)

Closes # (issue #)

### CODE REVIEW

File: odysseus_tree/configs/raspberrypi5_ap_defconfig [hunk 1]
```diff
@@ -79,3 +78,4 @@ BR2_PACKAGE_ALFA_LED=y
 BR2_PACKAGE_ALFA_LED_AP=y
 BR2_PACKAGE_RPI_DTLOADER=y
 BR2_PACKAGE_RPI_DTLOADER_FILES="$(BR2_EXTERNAL_ODY_TREE_PATH)/package/nrc7292/newracom_for_5-16_or_later-pi5.dts"
+BR2_PACKAGE_ODYSSEUS_DAEMON=y
```
Comment (caiodasilva2005) [inline odysseus_tree/configs/raspberrypi5_ap_defconfig:81]: why is the ap running odysseus daemon?

File: odysseus_tree/patches/linux/bba53a117a4a5c29da892962332ff1605990e17a/01-yeet_dma_align_check.patch
Comment (caiodasilva2005) [inline odysseus_tree/patches/linux/bba53a117a4a5c29da892962332ff1605990e17a/01-yeet_dma_align_check.patch:1]: why do we not care about alignment here?

File: odysseus_tree/sources/base_telemetry/publish.py
Comment (caiodasilva2005) [inline odysseus_tree/sources/base_telemetry/publish.py:1]: bye bye python

File: odysseus_tree/sources/wheel_buttons/src/main.c [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    rcfg = gpiod_request_config_new();
+    if (!set || !lcfg || !rcfg) goto out;
+
+    gpiod_line_settings_set_direction(set, GPIOD_LINE_DIRECTION_INPUT);
+    gpiod_line_settings_set_edge_detection(set, GPIOD_LINE_EDGE_RISING);
+    gpiod_line_settings_set_bias(set, GPIOD_LINE_BIAS_PULL_DOWN);
+    gpiod_line_settings_set_debounce_period_us(set, DEBOUNCE_US);
+
+    if (gpiod_line_config_add_line_settings(lcfg, offsets, n, set))
+        goto out;
```
Comment (caiodasilva2005) [inline odysseus_tree/sources/wheel_buttons/src/main.c:46]: crafty use of goto

File: odysseus_tree/sources/wheel_buttons/src/main.c [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        .fd     = gpiod_line_request_get_fd(req),
+        .events = POLLIN
+    };
+
+    printf("Monitoring %d buttons — Ctrl+C to exit\n", NUM_BUTTONS);
+
+    for (;;) {
+        if (poll(&pfd, 1, -1) < 0) {
+            if (errno == EINTR) continue;
+            perror("poll"); break;
```
Comment (caiodasilva2005) [inline odysseus_tree/sources/wheel_buttons/src/main.c:87]: what would happening in the case that we ever get an error here (or anything that breaks from loop)? does init.d have automatic restart of services when they go down? Cuz buttons would just stop working in this case

File: odysseus_tree/sources/wheel_buttons/src/main.c [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+            unsigned int gpio = gpiod_edge_event_get_line_offset(ev);
+
+            const ButtonMapping *btn = button_map_find(gpio);
+            if (!btn) continue;
+
+            can_send(can, btn);
+            printf("[CAN ] 0x%03X [%u %s] \n",
+                   CAN_ID, btn->index, btn->name);
+
+            uss_send(&unix_tx, &btn->index, sizeof(btn->index));
```
Comment (caiodasilva2005) [inline odysseus_tree/sources/wheel_buttons/src/main.c:106]: so this unix socket is for nero to just read directly for games and non-vcu dependent stuff correct? his way no latency from mqtt

File: odysseus_tree/configs/raspberrypi5_ap_defconfig (hunk 1 above)
Comment (jr1221) [reply odysseus_tree/configs/raspberrypi5_ap_defconfig:81]: Good question.  It is only running the --halow module which fetches the RSSI of the station (the signal strength of the halow connection).  This is for telemetry stand to do RSSI based aiming.

File: odysseus_tree/patches/linux/bba53a117a4a5c29da892962332ff1605990e17a/01-yeet_dma_align_check.patch
Comment (jr1221) [reply odysseus_tree/patches/linux/bba53a117a4a5c29da892962332ff1605990e17a/01-yeet_dma_align_check.patch:1]: Good question.  No idea.  I removed it and the HaLow started working.  I found it in a thread a while back on the raspberrypi/linux repo.

File: odysseus_tree/sources/wheel_buttons/src/main.c (hunk 3 above)
Comment (jr1221) [reply odysseus_tree/sources/wheel_buttons/src/main.c:87]: No there is no restarting of processes.  @AlexH-5736 can you just ignore the error here.

File: odysseus_tree/sources/wheel_buttons/src/main.c (hunk 4 above)
Comment (jr1221) [reply odysseus_tree/sources/wheel_buttons/src/main.c:106]: Correct.

**Comment by jr1221** [comment]
Merging now -- Any mapping updates can be conducted in followup PR.
